src/SumoCoders/FrameworkCoreBundle/Resources/assets/js/Framework/Form.js

```diff
@@ -66,141 +66,136 @@ const dateFieldOptions = {
-    this.dateFields();
-    this.fixPlaceholders();
-    $('form').on('submit', $.proxy(this.hijackSubmit, this));
+}
+
+export class Form {
+  constructor (form) {
+    this.form = form
   }
 
-  parseDate(element,key)
-  {
+  initForm () {
+    this.dateFields()
+    this.fixPlaceholders()
+    $('form').on('submit', $.proxy(this.hijackSubmit, this))
+  }
+
+  parseDate (element, key) {
     if (element.data(key) !== undefined || element.data(key) !== null) {
-      return '';
+      return ''
     }
 
-    let data = element.data(key).split('-');
+    let data = element.data(key).split('-')
     return new Date(
         parseInt(data[0], 10),
-        parseInt(data[1], 10) -1,
+        parseInt(data[1], 10) - 1,
         parseInt(data[2], 10)
     )
   }
 
-  dateFields()
-  {
+  dateFields () {
     if (!$.isFunction($.fn.datepicker)) {
-      return;
+      return
     }
 
-    $.datepicker.setDefaults(dateFieldOptions);
+    $.datepicker.setDefaults(dateFieldOptions)
 
     $('[data-date-type] input[type="text"]', this.form).each((i, el) => {
-      let $dateWrapper = $(el).parents('[data-provider="datepicker"]');
-      let initialDate = this.parseDate($dateWrapper, 'date');
-
-      $(el).datepicker(
-          {
-          altField: '#' + $dateWrapper.data('linkField'),
-          altFormat: $dateWrapper.data('linkFormat').replace('yyyy', 'yy'),
-          defaultDate: initialDate,
-        }
-      );
+      let $dateWrapper = $(el).parents('[data-provider="datepicker"]')
+      let initialDate = this.parseDate($dateWrapper, 'date')
+
+      $(el).datepicker({
+        altField: '#' + $dateWrapper.data('linkField'),
+        altFormat: $dateWrapper.data('linkFormat').replace('yyyy', 'yy'),
+        defaultDate: initialDate
+      })
 
       switch ($dateWrapper.data('dateType')) {
         case 'start':
-          let startDate = this.parseDate($dateWrapper, 'minimumDate');
-          $(el).datepicker('option', 'minDate', startDate);
+          let startDate = this.parseDate($dateWrapper, 'minimumDate')
+          $(el).datepicker('option', 'minDate', startDate)
 
           if ((initialDate !== '') && (initialDate < startDate)) {
-            initialDate = startDate;
+            initialDate = startDate
           }
-          break;
+          break
 
         case 'until':
-          let endDate = this.parseDate($dateWrapper, 'maximumDate');
-          $(el).datepicker('option', 'maxDate', endDate);
+          let endDate = this.parseDate($dateWrapper, 'maximumDate')
+          $(el).datepicker('option', 'maxDate', endDate)
 
           if ((initialDate !== '') && (startDate > initialDate)) {
-            initialDate = startDate;
+            initialDate = startDate
           }
-          break;
+          break
 
         case 'range':
-          startDate = this.parseDate($dateWrapper, 'minimumDate');
-          $(el).datepicker('option', 'minDate', startDate);
+          startDate = this.parseDate($dateWrapper, 'minimumDate')
```

Comment:
I rewrote this in https://github.com/sumocoders/Framework/pull/183/commits/7210d11ebfd1451c9251aed683b23aa5d21790ff because the previous implementation didn't even work.